### PR TITLE
Fix secure issue for ConfigTransportTLS

### PR DESCRIPTION
**Pull request by fisherxu**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
When insecure is true, defaultTLS should be false. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by fisherxu** [comment]
cc @liggitt @lavalamp

File: staging/src/k8s.io/client-go/rest/url_utils.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func defaultServerUrlFor(config *Config) (*url.URL, string, error) {
 	// config.Insecure is taken to mean "I want HTTPS but don't bother checking the certs against a CA."
 	hasCA := len(config.CAFile) != 0 || len(config.CAData) != 0
 	hasCert := len(config.CertFile) != 0 || len(config.CertData) != 0
-	defaultTLS := hasCA || hasCert || config.Insecure
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/rest/url_utils.go:87]: `config.Insecure = true` here means "use https but do not verify the certificate", not "use http"

**Comment by liggitt** [comment]
this is not correct, setting non-default options in TLSConfig implies a scheme-less server URL should use https

/close

File: staging/src/k8s.io/client-go/rest/url_utils.go (hunk 1 above)
Comment (fisherxu) [reply staging/src/k8s.io/client-go/rest/url_utils.go:87]: Okey, get it, thanks :)

**Pull request closed without merging**